compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/ToolAnnotationAnalysisTask.java

```diff
@@ -0,0 +1,196 @@
+        if (!(memberNode instanceof FunctionDefinitionNode functionDefinition)) {
+            return false;
+        }
+        return functionDefinition.functionBody() instanceof ExternalFunctionBodyNode externalFunctionBodyNode
+                && hasLlmCallAnnotation(externalFunctionBodyNode, npModulePrefixStr);
+    }
+
+    private void extractAndStoreSchemas(SemanticModel semanticModel, FunctionDefinitionNode functionDefinition,
+                                        Map<String, String> typeSchemas, TypeMapper typeMapper) {
+        Optional<ReturnTypeDescriptorNode> returnTypeNodeOpt = functionDefinition.functionSignature().returnTypeDesc();
+        if (returnTypeNodeOpt.isEmpty()) {
+            return;
+        }
+
+        ReturnTypeDescriptorNode returnTypeNode = returnTypeNodeOpt.get();
+        Optional<TypeSymbol> typeSymbolOpt = semanticModel.type(returnTypeNode.type().lineRange());
+        if (typeSymbolOpt.isEmpty()) {
+            return;
+        }
+
+        TypeSymbol typeSymbol = typeSymbolOpt.get();
+        if (typeSymbol instanceof UnionTypeSymbol unionTypeSymbol) {
+            for (TypeSymbol memberType : unionTypeSymbol.memberTypeDescriptors()) {
+                if (memberType instanceof TypeReferenceTypeSymbol typeReferenceTypeSymbol) {
+                    Schema schema = typeMapper.getSchema(typeReferenceTypeSymbol);
+                    typeSchemas.put(typeReferenceTypeSymbol.definition().getName().get(), getJsonSchema(schema));
+                }
+            }
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
+    private static String getJsonSchema(Schema schema) {
+        modifySchema(schema);
+        OpenAPISchema2JsonSchema openAPISchema2JsonSchema = new OpenAPISchema2JsonSchema();
+        openAPISchema2JsonSchema.process(schema);
+        String newLineRegex = "\\R";
+        String jsonCompressionRegex = "\\s*([{}\\[\\]:,])\\s*";
+        return Json.pretty(schema.getJsonSchema())
+                .replaceAll(newLineRegex, EMPTY_STRING)
+                .replaceAll(jsonCompressionRegex, "$1");
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static void modifySchema(Schema schema) {
+        if (schema == null) {
+            return;
+        }
+        modifySchema(schema.getItems());
+        modifySchema(schema.getNot());
+
+        Map<String, Schema> properties = schema.getProperties();
+        if (properties != null) {
+            properties.values().forEach(ToolAnnotationAnalysisTask::modifySchema);
+        }
+
+        List<Schema> allOf = schema.getAllOf();
+        if (allOf != null) {
+            schema.setType(null);
+            allOf.forEach(ToolAnnotationAnalysisTask::modifySchema);
+        }
+
+        List<Schema> anyOf = schema.getAnyOf();
+        if (anyOf != null) {
+            schema.setType(null);
+            anyOf.forEach(ToolAnnotationAnalysisTask::modifySchema);
+        }
```

Comment:
Why do we need to do this?

Reply:
Yes. OpenAPI module provides schemas that include unnecessary fields related to it. We need to filter out those fields.

Reply:
Why though? Our unions map to anyOf. Can we check if that still works properly?

Reply:
I discussed this with Subthar. The reason for doing this is that they have a specific structure if they don't set the type to null before updating, it results in an error.

Reply:
Can you check if the schema is generated correctly for unions?